src/app/workflow/workflow-detail/workflow-detail.component.ts

```diff
@@ -391,6 +349,114 @@ export class WorkflowDetailComponent implements OnInit, OnDestroy {
       });
   }
 
+
+  openCopy(content, jobId: string) {
+    this.getJob(content, jobId);
+  }
+
+  getJob(content, jobId: string ) {
+    this.jobService.getJob(jobId).subscribe(job => {
+        this.jobToCopy = job;
+        this.jobService.getPlugin(job.wippExecutable).subscribe(plugin => {
+          this.selectedSchema = this.pluginList.find(x => x.id === plugin.id);
+          this.jobModel['taskName'] = this.jobToCopy.name + '-copy';
+          this.jobModel['inputs'] = {};
+          const requests = [];
+          for (const input of Object.keys(this.jobToCopy.parameters)) {
+            if ((this.selectedSchema.properties.inputs.properties[input]['widget'] === 'search' || this.selectedSchema.properties.inputs.properties[input]['widget']['id'] === 'search') && this.jobToCopy.parameters[input].indexOf('{') === -1) {
+              const id = this.jobToCopy.parameters[input];
+
+              // Resolve AbstractFactory
+              const injectable = dataMap.get(this.selectedSchema.properties.inputs.properties[input]['format']);
+              // Inject service
+              this.service = this.injector.get(injectable);
+              requests.push(this.service.getById(id).pipe(map(response => {
+                  response['data'] = response;
+                  response['inputName'] = input;
+                  return response;
+                })
+              ));
+
+            }  else if (this.jobToCopy.parameters[input].indexOf('{') !== -1) {
+              this.jobModel['inputs'][input] = {};
+              this.jobModel['inputs'][input]['id'] = this.jobToCopy.parameters[input];
+              const prevId = this.jobToCopy.parameters[input].substring(3, this.jobToCopy.parameters[input].indexOf('.'));
+              const prevOutputName = this.jobToCopy.parameters[input].substring(this.jobToCopy.parameters[input].indexOf('.'), this.jobToCopy.parameters[input].length - 3);
+              const prevJob = this.jobs.find(x => x.id === prevId);
+              this.jobModel['inputs'][input]['name'] = prevJob.name + prevOutputName ;
+            } else {
+              this.jobModel['inputs'][input] = this.jobToCopy.parameters[input] ? this.jobToCopy.parameters[input] : null ;
+            }
+          }
+          if (requests.length === 0) {this.openCopyModal(content);
+          } else {forkJoin(requests).subscribe(results => {
+            for (const result of results) {
+              this.jobModel['inputs'][result['inputName']] = result['data'];
+            }
+            this.openCopyModal(content);
```

Comment:
You could just use the existing `open` function here, instead of creating a new one for copy